dkg/dkg.go

```diff
@@ -48,6 +53,104 @@ type Config struct {
 	TestSigning bool
 }
 
+type sigData struct {
+	duty   core.DutyType
+	pubkey core.PubKey
+	psigs  []core.ParSignedData
+}
+
+// exchanger is responsible for exchanging partial signatures between peers on libp2p.
+type exchanger struct {
+	sigChan  chan sigData
+	sigex    *parsigex.ParSigEx
+	sigdb    *parsigdb.MemDB
+	numPeers int
+	numVals  int
+}
+
+func newExchanger(tcpNode host.Host, peerIdx int, peers []peer.ID, vals int) *exchanger {
+	ex := &exchanger{
+		// threshold is len(peers) to wait until we get all the partial sigs from all the peers per DV
+		sigdb:    parsigdb.NewMemDB(len(peers)),
+		sigex:    parsigex.NewParSigEx(tcpNode, peerIdx, peers),
+		sigChan:  make(chan sigData, vals),
```

Comment:
this can be a blocking channel (len==0) since the `exchange` goroutine is reading from the channel when the parsigdb threshold is reached